language/move-vm/runtime/src/loader.rs

```diff
@@ -414,16 +429,40 @@ impl Loader {
         Ok(())
     }
 
-    fn load_module(
-        &self,
-        id: &ModuleId,
-        context: &dyn InterpreterContext,
-    ) -> VMResult<Arc<Module>> {
-        if let Some(module) = self.module_cache.lock().unwrap().get(id) {
-            return Ok(module);
+    pub(crate) fn check_natives(module: &VerifiedModule) -> VMResult<()> {
```

Comment:
this is the "linker" check. 
It checks this VM is exposing the given function as native.
In time we can use the signature in `Function` to verify arguments

Reply:
Maybe my confusion comes from when this is called. When is this invoked? Every time the module is loaded?

Reply:
every time the module is loaded and when a module is published.
I though I explained it but I see now I only did to myself ;)
- On publish, if a module has a native the VM/Runtime is asked if that native is expected. If not the module does not verify and does not publish. Implication is also that to add a native you have to version the VM before you can push the native
- On loading, a native is linked, that is, is resolved and the "function" cached away, so linked at load time as it should be.
The hope is that the combination of both gives us enough guarantees to feel good.

Linkage can and will be extended to some signature check. We can talk about ways to accomplish that, I have some idea but don't know if good or bad

Reply:
oh also, that is basically what happens now with the check in the verifier.
We are just making all the decision and semantic of natives part of the VM and not the verifier